### DISCUSSION TITLE
Solving Mixed-Integer + Semidefinite + Bilinear optimization problems

### DISCUSSION THREAD

**Discussion by NAMBQW** [yalmip/YALMIP#1332] [General discussion] [answered]
Hello,

I wonder if it is possible to solve a mixed integer semidefinite programming (MISDP) with some extra bilinear constraints by one of the internal solver in YALMIP? If yes, which solver can do it?

**Comment by johanlofberg** [comment]
Before you even think about that, you should think about if it is possible to reformulate to something where you don't have this product (i.e. , if you have a product between binaries and continuous, or if you can apply standard tricks such as Schur compolements).

If not, your only chance in YALMIP is to use BMIBNB and hope it works despite chances being close to 0. You are adviced to post the model though to see if there is anything to fix with the model because my default answer would be you will fail to solve this

**Reply by NAMBQW** [reply]
Thank you Prof. Johan for your answer. I read the explanation of the solver CUTSDP. I am new but if I understand well, the solver relaxes the semidefinite constraints to a finite number of linear constraints and the relaxation problem in this case is a mixed-integer bilinear programming which can be dealt by GUROBI. What I do not understand is how CUTSDP generates first vectors. I wonder with CUTSDP and appropriate external solvers, is that possible to solve this problem? Could you please explain to me a bit about the CUTSDP solver? Thank you very much.

**Reply by johanlofberg** [reply, marked as answer]
Anything can be used for the initial cut (if a cut on the SDP cone is needed in the first iteration due to unboundedness or something), such as setting all variables to 0 or what ever.
